test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/NodeFilterTests.cs

```diff
@@ -610,5 +613,211 @@ public void TextNumericFilters()
             };
             Assert.Throws<IndexOutOfRangeException>(() => INodeFilter.GetPropertyFilter(filterList, typeof(SqlHistoryTableQuerier), ValidForFlag.All), "Error not thrown when the array contains single value for between operator");
         }
+
+        [Test]
+        public void TestLikeFilter()
+        {
+            // Testing text filter with ends with operator
+            var filterList = new List<NodePropertyFilter>
+            {
+                new NodePropertyFilter()
+                {
+                    Property = "Name",
+                    Type = typeof(string),
+                    ValidFor = ValidForFlag.All,
+                    Values = new List<object> { "%test" },
+                    FilterType = FilterType.LIKE,
+                    IsDateTime = false
+                }
+            };
+            var filterString = INodeFilter.GetPropertyFilter(filterList, typeof(SqlHistoryTableQuerier), ValidForFlag.All);
+            Assert.AreEqual("[(like(@Name, '%test'))]", filterString, "Error parsing text filter with like operator");
```

Comment:
What about values entered by the user that have % in them?

Do you have tests that actually run the filter to verify that the generated filter string is correct?

Reply:
The like operator by default will have no wildcards. We take the string from the user and use it as it is.

Reply:
Yes, but did you test what happens if the user enters a string with a % in it? That's a valid character to have in object names.

Reply:
See https://github.com/microsoft/azuredatastudio/issues/23391 for a very similar issue

Reply:
The LIKE should mean (open to discussion) that the user would enter the wildcard.  So they could do

Order%
%Order%
%Order

Reply:
Hm, I still feel like this an awfully T-SQL specific way of doing this for something that's currently in the general extensibility layer. Will every provider then be expected to handle the same % syntax? Or is it up to them how they parse like statements?

Even for this - does the filtering correctly handle escaping the % then? e.g. if I enter `%[%]%` will it find all objects with % in their names?

Reply:
(I realize that would be a SMO issue if it didn't work if we're assuming the user has to know to enter valid syntax, but we should test it)

Reply:
Looking into it a bit more, it does at least seem that LIKE with % is a general SQL thing. So better. But then we're saying in ADS that every single other filter/search uses wildcards/regex but this one place uses SQL like? That seems weird and not very user friendly to me...could we not make it similar to the other filter widgets in the app (maybe only supporting wildcards though since regex isn't supported AFAIK) and at least use `*`?